sdk/samples/router/general.py

```diff
@@ -0,0 +1,320 @@
+                        f"deployed.")
+
+    # Wait for the dependencies of the first version to be fully undeployed
+    time.sleep(15)
+
+    # 6. Deploy a specific router config version (the first one we created)
+    response = my_router.deploy_version(first_ver_no)
+    print(f"6. You have deployed version {response['version']} of router {response['router_id']}.")
+
+    # Wait for the first version to get deployed
+    try:
+        my_router.wait_for_version_status(RouterStatus.DEPLOYED, first_ver_no)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} with version {first_ver_no} to get "
+                        f"deployed.")
+
+    # 7. Undeploy the current active router configuration
+    response = my_router.undeploy()
+    print(f"7. You have undeployed router {response['router_id']}.")
+
+    # Wait for the router to get undeployed
+    try:
+        my_router.wait_for_status(RouterStatus.UNDEPLOYED)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} to get undeployed.")
+
+    # Wait for the dependencies of the router to be fully undeployed
+    time.sleep(10)
+
+    # 8. Deploy the router's *current* configuration (notice how it still deploys the *first* version)
+    response = my_router.deploy()
+    print(f"8. You have deployed version {response['version']} of router {response['router_id']}.")
+
+    # Wait for the router to get deployed
+    try:
+        my_router.wait_for_status(RouterStatus.DEPLOYED)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} to get deployed.")
+
+    # Undeploy the router
+    response = my_router.undeploy()
+    print(f"You have undeployed router {response['router_id']}.")
+
+    # 9. Get a specific router version of the router
+    my_router_ver = my_router.get_version(first_ver_no)
+    print(f"9. You have just retrieved version {my_router_ver.version} of your router.")
+
+    # 10. Delete a specific router version of the router
+    response = my_router.delete_version(latest_ver_no)
+    print(f"10. You have deleted version {response['version']} of router {response['router_id']}.")
+
+    # 11. Get all deployment events associated with this router
+    events = my_router.get_events()
+    print(f"11. You have just retrieved a list of {len(events)} events for your router:")
+    for e in events:
+        print(e)
+
+    # 12. Delete this router (using its router_id)
+    deleted_router_id = turing.Router.delete(my_router.id)
+    print(f"12. You have just deleted the router with id: {deleted_router_id}")
+
+    # Check if the router still exists
+    for r in turing.Router.list():
+        if r.name == my_router_config.name:
+            raise Exception("Oh my, this router still exists!")
+
+
+if __name__ == '__main__':
```

Comment:
This is great!
Can we also explicitly cover (probably, in a separate file) the scenario, when user creates a NEW router, by using an existing router as a base and slightly tweaking its config? Something like:

```python
router_1 = turing.Router.get(my_router.id)

my_router_config = router_1.config

# Modify something in the router config
my_router_config.enricher.env.append(
        EnvVar(
                name="another",
                value="value"
            )
    )

router_2 = turing.Router.create(my_router_config)
```

Reply:
-Same comment as the one above-